### ISSUE TITLE
npm install of latest 5.0.0-rc0 fails against latest angular

### ISSUE THREAD

**Issue by bobtarling** [angular/material2#8263]
#### Bug, feature request, or proposal:
Since the very recent release of rc0 of 5.0.0 npm install is failing for me

#### What is the expected behavior?
npm install completes as it currently still does with 2.0.0-beta.12

#### What is the current behavior?

npm install --save @angular/material @angular/cdk

gives the following results

+-- @angular/cdk@5.0.0-rc0
+-- UNMET PEER DEPENDENCY @angular/common@4.4.6
+-- UNMET PEER DEPENDENCY @angular/core@4.4.6
+-- @angular/material@5.0.0-rc0
+-- UNMET PEER DEPENDENCY jquery@1.9.1 - 3
`-- UNMET PEER DEPENDENCY popper.js@^1.12.3

npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.1.2
npm WARN bootstrap@4.0.0-beta.2 requires a peer of jquery@1.9.1 - 3 but none was installed.
npm WARN bootstrap@4.0.0-beta.2 requires a peer of popper.js@^1.12.3 but none was installed.
npm WARN @angular/cdk@5.0.0-rc0 requires a peer of @angular/core@^5.0.0 but none was installed.
npm WARN @angular/cdk@5.0.0-rc0 requires a peer of @angular/common@^5.0.0 but none was installed.
npm WARN @angular/material@5.0.0-rc0 requires a peer of @angular/core@^5.0.0 but none was installed.
npm WARN @angular/material@5.0.0-rc0 requires a peer of @angular/common@^5.0.0 but none was installed.

#### What are the steps to reproduce?

npm install --save @angular/material @angular/cdk

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

5.0.0-rc0

#### Is there anything else we should know?

I am currently working round with

npm install --save @angular/material/2.0.0-beta.12 @angular/cdk/2.0.0-beta.12

**Comment by LukaszGrela** [comment]
I have similar experience, my output is as follows

```
+-- @angular/cdk@5.0.0-rc0
+-- UNMET PEER DEPENDENCY @angular/common@4.4.6
+-- UNMET PEER DEPENDENCY @angular/core@4.4.6
`-- @angular/material@5.0.0-rc0

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN @angular/material@5.0.0-rc0 requires a peer of @angular/core@^5.0.0 but none was installed.
npm WARN @angular/material@5.0.0-rc0 requires a peer of @angular/common@^5.0.0 but none was installed.
npm WARN @angular/cdk@5.0.0-rc0 requires a peer of @angular/core@^5.0.0 but none was installed.
npm WARN @angular/cdk@5.0.0-rc0 requires a peer of @angular/common@^5.0.0 but none was installed.

```

```

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/

Angular CLI: 1.5.0
Node: 6.11.5
OS: win32 x64
Angular: 4.4.6
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, tsc-wrapped

@angular/cli: 1.5.0
@angular-devkit/build-optimizer: 0.0.32
@angular-devkit/core: 0.0.20
@angular-devkit/schematics: 0.0.35
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.0
@schematics/angular: 0.1.1
typescript: 2.3.4
webpack: 3.8.1
```

**Comment by josephperrott** [comment]
Can you try entirely removing your `node_modules` and running `npm install` again?  I am not seeing the same issues myself and it looks like the dependency is updated in npm:

```
$ npm view @angular/cdk peerDependencies
{ '@angular/core': '^5.0.0',
  '@angular/common': '^5.0.0' }
```

```
$ npm view @angular/material peerDependencies

{ '@angular/cdk': '5.0.0-rc0',
  '@angular/core': '^5.0.0',
  '@angular/common': '^5.0.0' }
```

**Comment by josephperrott** [comment]
After another look, it appears that your applications are requiring angular `4.4.6`, rather than angular `5.0.0`.  With this `rc0` release, you will need to be using angular `5.0.0` as a peerDependency

**Comment by jelbourn** [comment]
FYI the published `package.json` has the right peerDependency version for Angular
https://unpkg.com/@angular/material@5.0.0-rc0/package.json

**Comment by zijianhuang** [comment]
I am already using Angular5:
```js
  "dependencies": {
    "@angular/animations": "^5.0.0",
    "@angular/cdk": "^2.0.0-beta.12",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/flex-layout": "^2.0.0-beta.10-4905443",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/material": "^5.0.0-rc0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/platform-server": "^5.0.0",
    "@angular/router": "^5.0.0",
    "@types/core-js": "^0.9.43",
    "@types/jasmine": "^2.6.3",
    "@types/node": "^8.0.49",
    "@types/selenium-webdriver": "^2.53.42",
    "concurrently": "^3.5.0",
    "core-js": "^2.5.1",
    "hammerjs": "^2.0.8",
    "moment": "^2.19.1",
    "reflect-metadata": "^0.1.10",
    "rimraf": "^2.6.2",
    "rxjs": "^5.5.2",
    "systemjs": "^0.20.19",
    "typescript": "^2.6.1",
    "zone.js": "^0.8.18"
  },
```

but npm insists on installing Material 12, since I had tried:
```
npm i @angular/material --save
npm WARN @angular/cdk@2.0.0-beta.12 requires a peer of @angular/core@~4.4.4 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/cdk@2.0.0-beta.12 requires a peer of @angular/common@~4.4.4 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/compiler-cli@5.0.0 requires a peer of typescript@>=2.4.2 <2.5 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/flex-layout@2.0.0-beta.10-4905443 requires a peer of @angular/core@~4.4.4 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/flex-layout@2.0.0-beta.10-4905443 requires a peer of @angular/common@~4.4.4 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/material@2.0.0-beta.12 requires a peer of @angular/core@~4.4.4 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/material@2.0.0-beta.12 requires a peer of @angular/common@~4.4.4 but none is installed. You must install peer dependencies yourself.
npm WARN tsickle@0.24.1 requires a peer of typescript@2.4.2 but none is installed. You must install peer dependencies yourself.
npm WARN angular-quickstart@1.0.0 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ @angular/material@2.0.0-beta.12
added 116 packages in 9.253s
```

or
npm install  with package.json above:

```
npm install
npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning EINTEGRITY: sha512-Y11dktBJ5YwKXX8DfHSb9V6shXKSHN5P2URPZLHTRjX3OsO/u8u1kZnSJvsuSH74aTg8f5ZKcpEeCdIJOcBkHg== integrity checksum failed when using sha512: wanted sha512-Y11dktBJ5YwKXX8DfHSb9V6shXKSHN5P2URPZLHTRjX3OsO/u8u1kZnSJvsuSH74aTg8f5ZKcpEeCdIJOcBkHg== but got sha1-ZdZGxejAzRve43BleZ+dPUh0glM=. (11465 bytes)
npm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.
npm WARN @angular/material@5.0.0-rc0 requires a peer of @angular/cdk@5.0.0-rc0 but none is installed. You must install peer dependencies yourself.
npm WARN angular-quickstart@1.0.0 No license field.

added 116 packages in 10.655s
```

**Comment by zijianhuang** [comment]
And this way seems to enable the installation of Material 5.0.0-rc0
Firstly run:
`npm i @angular/cdk@5.0.0-rc0 --save`

then run:
`npm i @angular/material@5.0.0-rc0 --save`

double checking the timestamps of the files reveals that they are the updated ones.

I had cleared the browser caches, when running my NG5 apps, I got this in browser's console:
```
zone.js:1057 GET http://localhost:28273/node_modules/@angular/common/bundles/common.umd.js/http 404 (Not Found)
(anonymous) @ zone.js:1057
fetchFetch @ fetch.js:32
(anonymous) @ instantiate.js:167
ZoneDelegate.invoke @ zone.js:392
Zone.run @ zone.js:142
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
resolvePromise @ zone.js:819
(anonymous) @ zone.js:746
Promise resolved (async)
(anonymous) @ zone.js:1049
ZoneAwarePromise @ zone.js:890
Ctor.then @ zone.js:1048
resolvePromise @ zone.js:799
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
resolvePromise @ zone.js:819
(anonymous) @ zone.js:746
Promise resolved (async)
(anonymous) @ zone.js:1049
ZoneAwarePromise @ zone.js:890
Ctor.then @ zone.js:1048
fetchFetch @ fetch.js:33
(anonymous) @ instantiate.js:167
ZoneDelegate.invoke @ zone.js:392
Zone.run @ zone.js:142
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
resolvePromise @ zone.js:819
(anonymous) @ zone.js:746
Promise resolved (async)
(anonymous) @ zone.js:1049
ZoneAwarePromise @ zone.js:890
Ctor.then @ zone.js:1048
resolvePromise @ zone.js:799
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
resolvePromise @ zone.js:819
(anonymous) @ zone.js:746
Promise resolved (async)
(anonymous) @ zone.js:1049
ZoneAwarePromise @ zone.js:890
Ctor.then @ zone.js:1048
fetchFetch @ fetch.js:33
(anonymous) @ instantiate.js:167
ZoneDelegate.invoke @ zone.js:392
Zone.run @ zone.js:142
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
resolvePromise @ zone.js:819
(anonymous) @ zone.js:746
Promise resolved (async)
(anonymous) @ zone.js:1049
ZoneAwarePromise @ zone.js:890
Ctor.then @ zone.js:1048
resolvePromise @ zone.js:799
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
resolvePromise @ zone.js:819
(anonymous) @ zone.js:746
Promise resolved (async)
(anonymous) @ zone.js:1049
ZoneAwarePromise @ zone.js:890
Ctor.then @ zone.js:1048
fetchFetch @ fetch.js:33
(anonymous) @ instantiate.js:167
ZoneDelegate.invoke @ zone.js:392
Zone.run @ zone.js:142
(anonymous) @ zone.js:873
ZoneDelegate.invokeTask @ zone.js:425
Zone.runTask @ zone.js:192
drainMicroTaskQueue @ zone.js:602
Promise resolved (async)
scheduleMicroTask @ zone.js:585
ZoneDelegate.scheduleTask @ zone.js:414
Zone.scheduleTask @ zone.js:236
Zone.scheduleMicroTask @ zone.js:256
scheduleResolveOrReject @ zone.js:871
ZoneAwarePromise.then @ zone.js:981
Loader.import @ loader-polyfill.js:64
SystemJSLoader$1.import @ systemjs-loader.js:126
(anonymous) @ (index):46
(index):46 Error: Fetch error: 404 Not Found
  Instantiating http://localhost:28273/node_modules/@angular/common/bundles/common.umd.js/http
  Loading http://localhost:28273/node_modules/@angular/material/bundles/material.umd.js
  Loading app
    at fetch.js:37
    at ZoneDelegate.invoke (zone.js:392)
    at Zone.run (zone.js:142)
    at zone.js:873
    at ZoneDelegate.invokeTask (zone.js:425)
    at Zone.runTask (zone.js:192)
    at drainMicroTaskQueue (zone.js:602)
    at <anonymous>
```

**Comment by zijianhuang** [comment]
I think that 5.0.0-rc0 is using HttpClient of NG5. So in systemjs.config.js, you need to add:
`            '@angular/common/http': 'npm:@angular/common/bundles/common-http.umd.js',`

Then the runtime will complain about some cdk components, then add these:
```
            '@angular/cdk/accordion': 'npm:@angular/cdk/bundles/cdk-accordion.umd.js',
            '@angular/cdk/layout': 'npm:@angular/cdk/bundles/cdk-layout.umd.js',
```

Then the runtime will complain:
`http://localhost:28273/node_modules/rxjs/operators/index.js/first.js 404 (Not Found)`

apparently this is related to a but in rxjs as documented at https://github.com/ReactiveX/rxjs/issues/2971

**Comment by zijianhuang** [comment]
Finally I got everything working with this 5.0.0-rc0. :) 

Step 0: You should have Angular 5 installed which works well with Material2 Beta 12.

Step 1: install cdk@5.0.0-rc0
`npm i @angular/cdk@5.0.0-rc0 --save`

Step2: install material 5
npm i @angular/material@5.0.0-rc0 --save

Step 3: declare HttpClient of NG5
In systemjs.config.js, add this:
`'@angular/common/http': 'npm:@angular/common/bundles/common-http.umd.js',`

This is for material 5 which apparently is using HttpClient, while your app code may still be using the Http service, so you may want to keep the declaration for the Http service.

Step 4: declare the following to address a bug in rxjs 5.5.0 - 5.5.2 which NG5 and Material 5 depend on.

```js
            //temp fix according to https://github.com/ReactiveX/rxjs/issues/2971
            'rxjs/operators': 'npm:rxjs/operators/index.js',
            'rxjs/operators/filter': 'npm:rxjs/operators/filter.js',
            'rxjs/operators/switchMap': 'npm:rxjs/operators/switchMap.js',
            'rxjs/operators/first': 'npm:rxjs/operators/first.js',
            'rxjs/operators/tap': 'npm:rxjs/operators/tap.js',
            'rxjs/operators/delay': 'npm:rxjs/operators/delay.js',
            'rxjs/operators/startWith': 'npm:rxjs/operators/startWith.js',
            'rxjs/operators/catchError': 'npm:rxjs/operators/catchError.js',
            'rxjs/operators/map': 'npm:rxjs/operators/map.js',
            'rxjs/operators/share': 'npm:rxjs/operators/share.js',
            'rxjs/operators/finalize': 'npm:rxjs/operators/finalize.js',
            'rxjs/operators/takeUntil': 'npm:rxjs/operators/takeUntil.js',
            'rxjs/operators/combineLatest': 'npm:rxjs/operators/combineLatest.js',
            'rxjs/operators/debounceTime': 'npm:rxjs/operators/debounceTime.js',
            'rxjs/operators/auditTime': 'npm:rxjs/operators/auditTime.js',
 
            'tslib' : 'npm:tslib/tslib.js'
```

When testing, make sure you have proper versioning of your code and meta, or you may clear the browser caches.

**Remarks:**
If you are using angular flex-layout, as of today, there seems to be a random bug during npm install or update, 2.0.0-rc.1 got installed instead. So goto something like YourNG2App\node_modules\@angular\flex-layout and check package.json to see if it is actually “@angular/flex-layout@2.0.0-rc.1”, if it is, do the following:
1.	Delete folder flex-layout
2.	Run `npm i @angular/flex-layout ` or `npm i @angular/flex-layout@2.0.0-beta.10
`

**Comment by aamir1995** [comment]
@zijianhuang are you using [angular-seed](https://github.com/mgechev/angular-seed)?

**Comment by zijianhuang** [comment]
@aamir1995 , my app had started from NG2 since last September. The issue is not related to how you initiate the app codebase, but related to on-going bugs and some hiccups of npm packages.

**Comment by zijianhuang** [comment]
(cont.)
If you are using Angular 5, Material2 5.0.0-rc0 and Angular Flex-layout, the following may work for you:

In package.json

```js
  "dependencies": {
    "@angular/animations": "^5.0.0",
    "@angular/cdk": "^5.0.0-rc0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/flex-layout": "^2.0.0-beta.10",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/material": "^5.0.0-rc0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/platform-server": "^5.0.0",
    "@angular/router": "^5.0.0",
    "@types/core-js": "^0.9.43",
    "@types/jasmine": "^2.6.3",
    "@types/node": "^8.0.49",
    "@types/selenium-webdriver": "^2.53.42",
    "concurrently": "^3.5.0",
    "core-js": "^2.5.1",
    "hammerjs": "^2.0.8",
    "moment": "^2.19.1",
    "reflect-metadata": "^0.1.10",
    "rimraf": "^2.6.2",
    "rxjs": "^5.5.2",
    "systemjs": "^0.20.19",
    "typescript": "^2.6.1",
    "zone.js": "^0.8.18"
  },
```

In systemjs.config.js

```js
    System.config({
        paths: {
            // paths serve as alias
            'npm:': 'node_modules/'

        },
        // map tells the System loader where to look for things
        map: {
            // our app is within the app folder
            app: 'app',
            //clientapi: 'clientapi',

            // angular bundles
            '@angular/core': 'npm:@angular/core/bundles/core.umd.js',
            '@angular/common': 'npm:@angular/common/bundles/common.umd.js',
            '@angular/compiler': 'npm:@angular/compiler/bundles/compiler.umd.js',
            '@angular/platform-browser': 'npm:@angular/platform-browser/bundles/platform-browser.umd.js',
            '@angular/platform-browser-dynamic': 'npm:@angular/platform-browser-dynamic/bundles/platform-browser-dynamic.umd.js',
            '@angular/flex-layout': 'npm:@angular/flex-layout/bundles/flex-layout.umd.js',
            '@angular/http': 'npm:@angular/http/bundles/http.umd.js',
            '@angular/common/http': 'npm:@angular/common/bundles/common-http.umd.js',
            '@angular/router': 'npm:@angular/router/bundles/router.umd.js',
            '@angular/forms': 'npm:@angular/forms/bundles/forms.umd.js',
            '@angular/material': 'npm:@angular/material/bundles/material.umd.js',
            '@angular/animations': 'npm:@angular/animations/bundles/animations.umd.js',
            '@angular/animations/browser': 'npm:@angular/animations/bundles/animations-browser.umd.js',
            '@angular/platform-browser/animations': 'npm:@angular/platform-browser/bundles/platform-browser-animations.umd.js',

            '@angular/cdk': 'npm:@angular/cdk/bundles/cdk.umd.js',
            '@angular/cdk/a11y': 'npm:@angular/cdk/bundles/cdk-a11y.umd.js',
            '@angular/cdk/bidi': 'npm:@angular/cdk/bundles/cdk-bidi.umd.js',
            '@angular/cdk/coercion': 'npm:@angular/cdk/bundles/cdk-coercion.umd.js',
            '@angular/cdk/collections': 'npm:@angular/cdk/bundles/cdk-collections.umd.js',
            '@angular/cdk/keycodes': 'npm:@angular/cdk/bundles/cdk-keycodes.umd.js',
            '@angular/cdk/observers': 'npm:@angular/cdk/bundles/cdk-observers.umd.js',
            '@angular/cdk/overlay': 'npm:@angular/cdk/bundles/cdk-overlay.umd.js',
            '@angular/cdk/platform': 'npm:@angular/cdk/bundles/cdk-platform.umd.js',
            '@angular/cdk/portal': 'npm:@angular/cdk/bundles/cdk-portal.umd.js',
            '@angular/cdk/rxjs': 'npm:@angular/cdk/bundles/cdk-rxjs.umd.js',
            '@angular/cdk/scrolling': 'npm:@angular/cdk/bundles/cdk-scrolling.umd.js',
            '@angular/cdk/stepper': 'npm:@angular/cdk/bundles/cdk-stepper.umd.js',
            '@angular/cdk/table': 'npm:@angular/cdk/bundles/cdk-table.umd.js',
            '@angular/cdk/accordion': 'npm:@angular/cdk/bundles/cdk-accordion.umd.js',
            '@angular/cdk/layout': 'npm:@angular/cdk/bundles/cdk-layout.umd.js',

            // other libraries
            'rxjs': 'npm:rxjs',
            //'angular-in-memory-web-api': 'npm:angular-in-memory-web-api/bundles/in-memory-web-api.umd.js',
            'moment': 'npm:moment/moment.js',

            //temp fix according to https://github.com/ReactiveX/rxjs/issues/2971
            'rxjs/operators': 'npm:rxjs/operators/index.js',
            'rxjs/operators/filter': 'npm:rxjs/operators/filter.js',
            'rxjs/operators/switchMap': 'npm:rxjs/operators/switchMap.js',
            'rxjs/operators/first': 'npm:rxjs/operators/first.js',
            'rxjs/operators/tap': 'npm:rxjs/operators/tap.js',
            'rxjs/operators/delay': 'npm:rxjs/operators/delay.js',
            'rxjs/operators/startWith': 'npm:rxjs/operators/startWith.js',
            'rxjs/operators/catchError': 'npm:rxjs/operators/catchError.js',
            'rxjs/operators/map': 'npm:rxjs/operators/map.js',
            'rxjs/operators/share': 'npm:rxjs/operators/share.js',
            'rxjs/operators/finalize': 'npm:rxjs/operators/finalize.js',
            'rxjs/operators/takeUntil': 'npm:rxjs/operators/takeUntil.js',
            'rxjs/operators/combineLatest': 'npm:rxjs/operators/combineLatest.js',
            'rxjs/operators/debounceTime': 'npm:rxjs/operators/debounceTime.js',
            'rxjs/operators/auditTime': 'npm:rxjs/operators/auditTime.js',
 
            'tslib' : 'npm:tslib/tslib.js'

        },

```

Then run `npm install`

And if you are using Angular flex-layout, please check the remarks above.

**Comment by bobtarling** [comment]
It looks like my original report is false.
My confusion as an angular (and npm) newbie is the process of upgrading from ng4 to ng5 and it seems the instructions I did fnd didn't work correctly.
The getting started guide is so good its a shame there upgrade process is not so easy to find.
Sorry if this distracted - for me this issue can be closed.

**Comment by zijianhuang** [comment]
@bobtarling , the getting started guide is good for getting started. If you have been developing along with Angular 2/4/5 and Material2 Beta 12, when upgrading, you may see a lot complains from various dependencies. Hopefully NG and MD teams will fix the defects in next release.

**Comment by Longfld** [comment]
@zijianhuang would you mind post your full package.json and systemjs.config.js here,  following your instructions, still fails

**Comment by zijianhuang** [comment]
@Longfld , what I had posted is almost the whole package.json and systemjs.config.js. Have you cleared the browser cache?

If you are not sure, try to check all checkboxes of clearing history. If you are using Windows and Visual Studio ASP.NET, you may clear %temp% and C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Temporary ASP.NET Files etc.

And in my own dev machine, if I launch the NG2 WebApp using VS 2017 IDE, IIS and Chrome, Chrome's console is still giving various error messages. However, with IE, FF and Edge, no problem. Strange enough, if I launch IIS Express hosting the app without VS IDE, Chrome just run my app correctly. And if I copy all copy all codes to local IIS, and the test server IIS, all things on all browsers are just working fine. And other dev machines has no such problem.

In short, cache could be the cause of the problem. In addition to the caches I had mentioned, VS IDE and Chrome together had used some old stuffs cached somewhere I haven't yet known. 

So you may test with other browsers in other environments to rule out the interruption from mysterious cache. I may be writing a blog soon when I have some hours, also talk about versioning of files.

**Comment by josephperrott** [comment]
Closing as the `npm install` issue in the posted issue was solved.

**Comment by Gusachenko** [comment]
Hi!
I have the same problem after update to new angular 5+ version.
Stack: **angular cli** + **angular material**.

Old package.json
```json
{
  "name": "ng-material-starter",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "4.1.0",
    "@angular/common": "4.1.0",
    "@angular/compiler": "4.1.0",
    "@angular/core": "4.1.0",
    "@angular/forms": "4.1.0",
    "@angular/http": "4.1.0",
    "@angular/material": "2.0.0-beta.5",
    "@angular/platform-browser": "4.1.0",
    "@angular/platform-browser-dynamic": "4.1.0",
    "@angular/router": "4.1.0",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "material-design-icons": "^3.0.1",
    "normalize.css": "^7.0.0",
    "roboto-fontface": "^0.7.0",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cdk": "^2.0.0-beta.8",
    "@angular/cli": "^1.1.2",
    "@angular/compiler-cli": "4.1.0",
    "@angular/language-service": "4.1.0",
    "@angular/service-worker": "^1.0.0-beta.15",
    "@types/jasmine": "2.5.45",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
}
```


New `angular  cli` + `angular material`  **5+ version**
```json
{
  "name": "ng-material-starter",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.0.0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/router": "^5.0.0",
    "@angular/material": "2.0.0-beta.5",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "material-design-icons": "^3.0.1",
    "normalize.css": "^7.0.0",
    "roboto-fontface": "^0.7.0",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "^1.5.0",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/language-service": "^4.2.4",
    "@angular/service-worker": "^1.0.0-beta.15",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@types/jasmine": "2.5.45",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.1.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.4.2"
  }
}
```

**Important**: when you change package.json file, don't forget for remove `node_modules` folder and then execute `npm install`.

**Comment by zijianhuang** [comment]
This issue was closed. I had mentioned I might will be writing a blog based on my responses here. Published a few days ago, here you are: https://www.codeproject.com/Tips/1213737/Upgrade-to-Angular-and-HttpClient

**Comment by srinivas444** [comment]
By Installing the failed UNMET dependencies globally one by one will help you, Because it's worked for me.

example :  npm install -g @angular/common@4.4.6.

**Comment by joseandrespena** [comment]
This issue is not fixed... I tried to install MEAN stack and there is an error... not sure where to log it..
